python/lzreposync/src/lzreposync/primary_parser.py

```diff
@@ -210,6 +212,11 @@ def parse_primary(self):
                 ):
                     # New package
                     doc.expandNode(node)
+
+                    arch_node = node.getElementsByTagName("arch")[0]
+                    pkg_arch = get_text(arch_node)
+                    if not re.fullmatch(self.arch_filter, pkg_arch):  # Filter by arch
```

Comment:
You would need to add `noarch` to the filter as that should be included in all cases

Reply:
done